### DISCUSSION TITLE
Any plan changing `highlightColor` to accessor?

### DISCUSSION THREAD

**Discussion by moong23** [visgl/deck.gl#8988] [Q&A] [answered]
According to the current docs, `highlightColor` seems like it only gets [Color | Function].
The reason i need `highlightColor` to be accessor is that I've combined several datas(like 3-4 path's point coordinate data) to calculate individual point's distance between others.
After that, if i want to highlight the color differently depends on the pathId, i need to use either
1. render new Layer whenever hovering point has changed (saving hovered point at state, re-render layer whenever state changes)
2. make layer as [ layer : path = N : N ]. This way can implement the feature without using state & tremendous re-rendering issue.

But I've thought that it could be much easier & simpler if `highlightColor` can be used as [Color | Function | Accessor].
So I'd like to know how you think about this feature.

FYI, I'm using React, Mapbox, Deck.gl

**Comment by Pessimistress** [comment]
I don't understand the ask. Why can't you achieve what you described by supplying a function to `highlightColor`?

**Reply by moong23** [reply]
Sry for the poor explanation :(

To achieve the feature I've wanted, I need to use a code like
```
highlightColor: (d) => d.color
```
since single Path data has different initial color and want to display different color to be displayed on hover with a data provided as 1-d array format.

**Reply by moong23** [reply]
For better explanation, here an example of the data

```
const data  = [
{
  id: "trajectoryA-0000001",
  trajectoryId : "trajectoryA",
  position : [number, number, number],
},
{
  id: "trajectoryA-0000002",
  trajectoryId : "trajectoryA",
  position : [number, number, number],
},
...
{
  id: "trajectoryB-0000001",
  trajectoryId : "trajectoryB",
  position : [number, number, number],
},
{
  id: "trajectoryB-0000001",
  trajectoryId : "trajectoryB",
  position : [number, number, number],
},
...
{
  id: "trajectoryC-0000001",
  trajectoryId : "trajectoryC",
  position : [number, number, number],
},
...
]

**Reply by Pessimistress** [reply, marked as answer]
https://deck.gl/docs/api-reference/core/layer#highlightcolor

> If a function is supplied, it is called with a [pickingInfo](https://deck.gl/docs/developer-guide/interactivity#the-pickinginfo-object) object when the hovered object changes. The return value is used as the highlight color for the picked object. Only works with autoHighlight: true.
